File: week 6/sentimental-mario-more/mario.py
while True:
    try:
        height = int(input("Height of the pyramid: "))
        if 1 <= height <= 8:
            break
    except ValueError:
        print("Invalid input. Please enter a number between 1 and 8.")

for rows in range(height):
    # print spaces
    for spaces in range(height - rows - 1):
        print(" ", end="")

    # print blocks
    for columns in range(rows + 1):
        print("#", end="")

    # prints spaces
    print("  ", end="")

    # prints blocks
    for columns in range(rows + 1):
        print("#", end="")

    # moves to a new line
    print()

File: week 1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // prompting the user for height
    int height;
    do
    {
        height = get_int("height of the pyramid: ");
    }
    while (height < 1 || height > 8);

    // print pyramid
    for (int rows = 0; rows < height; rows++)
    {
        // print spaces
        for (int spaces = 0; spaces < height - rows - 1; spaces++)
        {
            printf(" ");
        }

        // print blocks
        for (int columns = 0; columns <= rows; columns++)
        {
            printf("#");
        }

        // prints spaces
        printf("  ");

        // prints blocks
        for (int columns = 0; columns <= rows; columns++)
        {
            printf("#");
        }

        // moves to new line
        printf("\n");
    }
}
